pkg/apis/provisioning.cattle.io/v1/cluster.go

```diff
@@ -8,64 +8,136 @@ import (
 )
 
 // +genclient
-// +kubebuilder:skipversion
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=clusters,scope=Namespaced,categories=provisioning
+// +kubebuilder:subresource:status
```

Comment:
I know this is a draft PR, but wanted to call out that when working on the Node Driver Public API PR adding status as a `subresource` actually prevented status conditions from being shown in my debugger. We might want to look into any other potential side effects of setting this on existing CRDs

Reply:
When testing the node driver PR I added a breakpoint [here](https://github.com/rancher/rancher/blob/main/pkg/controllers/management/drivers/nodedriver/machine_driver.go#L85) and walked through the logic when enabling / disabling drivers in the ui and `kubectl`. When I was doing this, I could not see any of the values set in the status using the Goland debugger UI if the CRD included `//+kubebuilder:subresource:status`

Reply:
We may need to check and make sure the node driver objects are working correctly then, might be an issue because of norman.

Reply:
Well I omitted the `subresource:status` marker from that PR after noticing this behavior, and didn't find any issues debugging in my IDE or when testing functionality via the UI and cli. QA is going to pick that issue up soon and will do even more thorough testing.